sql: fix super columns for user/roles that have admin role

mnovelodou opened the pull request:
Previously, pg_authid.rolsuper, pg_roles.rolsuper and pg_user.usesuper
only were true if it was root or admin role
This was inadequate because we can have other users/roles with admin
role
To address this, this patch change mentioned columns to show true if a
user is part of admin role.

Release justification: low risk, high benefit changes to existing functionality
Release note (sql change): pg_authid.rolesuper, pg_roles.rolesuper and
pg_user.usesuper now are true for users/roles that have admin role

Fixes: #69943

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/69981)
<!-- Reviewable:end -->

cockroach-teamcity commented:
[![CLA assistant check](https://cla.crdb.dev/pull/badge/not_signed)](https://cla.crdb.dev/cockroachdb/cockroach?pullRequest=69981) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.crdb.dev/cockroachdb/cockroach?pullRequest=69981) before we can accept your contribution.<br/><hr/>**MiguelNovelo** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.crdb.dev/check/cockroachdb/cockroach?pullRequest=69981) it.</sub>

mnovelodou committed:
sql: fix super columns for user/roles that have admin role

Previously, pg_authid.rolsuper, pg_roles.rolsuper and pg_user.usesuper
only were true if it was root or admin role
This was inadequate because we can have other users/roles with admin
role
To address this, this patch change mentioned columns to show true if a
user is part of admin role.

Release justification: low risk, high benefit changes to existing functionality
Release note (sql change): pg_authid.rolesuper, pg_roles.rolesuper and
pg_user.usesuper now are true for users/roles that have admin role

Fixes: #69943

rafiss reviewed (approved):
lgtm!

rafiss commented:
bors r+

nvb reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/69981)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/sql/pg_catalog.go, line 528 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/69981#-MjbaUwpATT5d2sG04PF:-MjbaUwpATT5d2sG04PG:b-8lfx8f) ([raw file](https://github.com/cockroachdb/cockroach/blob/ecad54190d6eb1eb01193cd40bb2a0bc3b59938b/pkg/sql/pg_catalog.go#L528)):*
> ```Go
> 				h.UserOid(username),                  // oid
> 				tree.NewDName(username.Normalized()), // rolname
> 				tree.MakeDBool(isRoot || isSuper),    // rolsuper
> ```

drive-by comment: is `isRoot` needed here anymore? `userIsSuper` calls into `UserHasAdminRole`, which checks `user.IsRootUser() || user.IsNodeUser()`, so I think it's redundant. That doesn't mean what we have here is wrong, but is it redundant and potentially misleading?


<!-- Sent from Reviewable.io -->
